library/core/src/intrinsics.rs

```diff
@@ -853,7 +853,7 @@ extern "rust-intrinsic" {
     /// This will statically either panic, or do nothing.
     ///
     /// This intrinsic does not have a stable counterpart.
-    #[rustc_const_unstable(feature = "const_assert_type", issue = "none")]
+    #[rustc_const_stable(feature = "const_assert_type", since = "1.58.0")]
```

Comment:
We might as well const-stabilize the other two `assert` intrinsics as well? We don't need to but it seems more consistent.

Reply:
If there are no blockers I can do so.

Reply:
These intrinsics are implemented very similarly so I can't see how one would be blocked and the others not.

Reply:
Actually I take that back, the intrinsics are currently implemented in Miri only, not in CTFE.

I'll plan to send a PR implementing and const-stabilizing the intrinsics once this one lands, just for consistency.